packages/fxa-settings/src/lib/context/implementations/url-context.ts

```diff
@@ -0,0 +1,48 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { Context } from "..";
+
+export class UrlContext implements Context {
+
+  get pathName() {
+    return this.window.location.pathname;
+  }
+
+  constructor(
+    public readonly window: Window
+  ) {
+  }
+
+  get(key: string) {
+    const params = new URLSearchParams(this.window.location.search);
+    const rawValue = params.get(key);
+    if (rawValue == null) {
+      return null;
+    }
+
+    return window.decodeURIComponent(rawValue);
```

Comment:
We might need introduce some JSON serialization here. I'll address this later when tests are added.